src/trackinfoobject.cpp

```diff
@@ -147,28 +161,10 @@ void TrackInfoObject::setMetadata(const Mixxx::TrackMetadata& trackMetadata) {
     }
 }
 
-void TrackInfoObject::getMetadata(Mixxx::TrackMetadata* pTrackMetadata) const {
-    // TODO(XXX): This involves locking the mutex for every setXXX
-    // method. We should figure out an optimization where there are private
-    // getters that don't lock the mutex.
-    pTrackMetadata->setArtist(getArtist());
-    pTrackMetadata->setTitle(getTitle());
-    pTrackMetadata->setAlbum(getAlbum());
-    pTrackMetadata->setAlbumArtist(getAlbumArtist());
-    pTrackMetadata->setYear(Mixxx::TrackMetadata::reformatYear(getYear()));
-    pTrackMetadata->setGenre(getGenre());
-    pTrackMetadata->setComposer(getComposer());
-    pTrackMetadata->setGrouping(getGrouping());
-    pTrackMetadata->setComment(getComment());
-    pTrackMetadata->setTrackNumber(getTrackNumber());
-    pTrackMetadata->setTrackTotal(getTrackTotal());
-    pTrackMetadata->setChannels(getChannels());
-    pTrackMetadata->setSampleRate(getSampleRate());
-    pTrackMetadata->setDuration(getDuration());
-    pTrackMetadata->setBitrate(getBitrate());
-    pTrackMetadata->setReplayGain(getReplayGain());
-    pTrackMetadata->setBpm(Mixxx::Bpm(getBpm()));
-    pTrackMetadata->setKey(getKeyText());
+void TrackInfoObject::getTrackMetadata(Mixxx::TrackMetadata* pTrackMetadata, bool* pParsedFromFile) const {
+    QMutexLocker lock(&m_qMutex);
+    *pTrackMetadata = m_metadata;
+    *pParsedFromFile = m_bHeaderParsed;
```

Comment:
Now the pParsedFromFile flag looks like to indicate that the Metadata is 100 % from the file. 

"m_bHeaderParsed" distingushes a file with initial metadata from a the file, from a file that has no file meta data. 
Even if a users has changes all metadata, m_bHeaderParsed is still true.  

How about just keep the name  "pHeaderParsed" to match the library name or something like "pParsedFromFileBefore"?